### ISSUE TITLE
Implement simple PipelineRun

### ISSUE THREAD

**Issue by bobcatfish** [knative/build-pipeline#61]
## Expected Behavior
A user should be able to create a `PipelineRun` which will look at the `Tasks` in the referenced `Pipeline` and for each of those, create a `TaskRun`.

If #59 is done before this, creation of the `TaskRun` would also trigger running of the `Task`, however if not, nothing would actually be executed but the `TaskRuns` would exist.

Out of scope of this issue (coming in later issues):
* Providing inputs to the Tasks
* Doing something with produced outputs

Requirements:
* Test coverage (unit tests + if possible integration tests, may be blocked by #16 )
* Docs that explain how to create a PipelineRun, with examples
* Docs that explain how to debug this (i.e. where to look if it goes wrong)

## Actual Behavior
If you create a `PipelineRun`, nothing will happen.

## Steps to Reproduce the Problem

  1. Deploy the Pipeline CRD + related CRDs to your cluster (note: should be docs on how to do this post kubebuilder)
  2. Create a `Task` like this:
```yaml
apiVersion: pipeline.knative.dev/v1beta1
kind: Task
metadata:
  name: helloworld-task
  namespace: default
spec:
    buildSpec:
        steps:
            - name: helloworld
              image: busybox
              args: ['echo', 'hello', 'build']
```
  3. Create a `Pipeline` like this:

```yaml
apiVersion: pipeline.knative.dev/v1beta1
kind: Pipeline
metadata:
  name: hello-pipeline
  namespace: default
spec:
    tasks:
        - name: helloworld-task-1
          taskRef:
              name: helloworld-task
        - name: helloworld-task-2
          taskRef:
              name: helloworld-task
        - name: helloworld-task-3
          taskRef:
              name: helloworld-task
```

4. Create a `PipelineRun` like this:

```yaml
apiVersion: pipeline.knative.dev/v1beta1
kind: PipelineRun
metadata:
  name: hello-pipeline-run-1
  namespace: default
spec:
    pipelineRef:
        name: hello-pipeline
    triggerRef:
        type: manual
```

5. Creating the PipelineRun should have created `TaskRuns`, so we should then be able to see:

  a. The `status` of the `PipelineRun` should be updated (see [example](https://github.com/knative/build-pipeline/blob/master/examples/invocations/kritis-pipeline-run.yaml#L27)), check with `kubectl get pipelineRuns`
  b. There should be 3 `TaskRuns` created, each with a unique name and a reference to the `PipelineRun` (check with `kubectl get taskRuns`), for example:

```yaml
apiVersion: pipeline.knative.dev/v1beta1
kind: TaskRun
metadata:
  name: helloworld-task-run-1-1
spec:
    taskRef:
        name: helloworldTask
    trigger:
        triggerRef:
            type: manual
```

(I left the PipelineParams reference out of the PipelineRun, but if it turns out that we need a ServiceAccount to make this work, we'll need to specify it in the PipelineParams and reference it in the PipelineRun as well)

**Comment by tejal29** [comment]
@bobcatfish Was looking at if there is a way to divide this further.
Few subtasks i am thinking of:
- [ ] Add unit tests for task_controller.go
- [ ] Add unit tests for task_run_controller.go

On the branch, i went ahead and added all the methods 
- runWorker
- processNextWorkItem
- enqueueBuild
We could potentially see if 
https://github.com/knative/build-pipeline/blob/master/pkg/controller/taskrun/controller.go#L275 has  some tasks recorded.
